server/routes/updateAddress.ts

```diff
@@ -0,0 +1,55 @@
+import { Request, Response, NextFunction } from 'express';
+import { factory, formatFilename } from '../../shared/logging';
+import { DB, sequelize } from '../database';
+import { Errors } from "./verifyAddress";
+
+const log = factory.getLogger(formatFilename(__filename));
+
+// update ghost address for imported discourse users by new address
+const updateAddress = async (models: DB, req: Request, res: Response, next: NextFunction) => {
+    const { address, chain } = req.body
+    if (!address) {
+        return next(new Error(Errors.NoAddress));
+    }
+    if (!chain) {
+        return next(new Error(Errors.NoChain));
+    }
+    const { id: ghostAddressId } = await models.Address.scope('withPrivateData').findOne({
+        where: { chain, ghost_address: true, user_id: req.user.id }
```

Comment:
We should ensure `req.user.id` isn't null/undefined before doing this query.

Reply:
@jnaviask I added a test to make sure `req.user.id` is not null